interview-questions.md

```diff
@@ -64,6 +64,6 @@ Research and define the following terms to the best of your ability.
 
 3. React state: A React object that is built-in and used to contain data or information about the component.
 
-4. React lifecycle methods: The three categories of lifecycle methods are: mounting, updating, and unmounting.
+4. React lifecycle methods: Special methods that are built into React to be used on components while in the DOM.
 
-5. DOM: This is a website platform and language-neutral interface that gives programs and scripts the ability to dynamically access and update a documents structure, content, and style.
+5. DOM: A website platform and language-neutral interface that gives programs and scripts the ability to dynamically access and update documents structure, content, and style.
```

Comment:
Excellent answers in this section.  Just a note on "What is the difference between map and filter?" - map and filter are only ran on arrays not objects.  Map will create a new array the same length as the original array and filter will return an array smaller than the original array based on specific conditions.